aaelf64-morello/aaelf64-morello.rst

```diff
@@ -252,6 +256,37 @@ the same name in `AAELF64`_:
 Object Files
 ============
 
+ELF Header
+----------
+
+The ELF header provides a number of fields that assist in interpretation of the
+file. Most of these are specified in the base standard. The following fields
+have Morello-specific meanings.
+
+``e_flags``
+  The processor-specific flags are shown in the following table.
+
+.. _Morello-specific e_flags:
+
+.. class:: aaelf64-morello-elf-flags
+
+.. table:: Morello-specific e_flags
+
+  +---------------------------------+--------------------------------------------------------------+
+  | Value                           | Meaning                                                      |
+  +---------------------------------+--------------------------------------------------------------+
+  | :code:`EF_AARCH64_MORELLO_EXT`  | The ELF file contains uses an ABI where all pointers are     |
+  | (:code:`0x00010000`)            | implemented using capabilities (Pure-capability ABI)         |
+  +---------------------------------+--------------------------------------------------------------+
+
+Sections
+--------
+
+Special Sections
+^^^^^^^^^^^^^^^^
+
+A Morello toolchain can emit ELF Note sections in accordance to [CHERI_ELF_].
```

Comment:
Should Morello prove successful, do you want to reserve the ability to reuse this flag to be its real non-prototype successor? As it stands this bit will forever be wasted once the Morello program comes to an end.

Reply:
i.e. should this really be `EF_AARCH64_CHERI_PURECAP`?

Reply:
That is my only concern as well and the reason I had originally suggested an ELF note with a name of "Morello" as it could be easily jettisoned in the future.  Checking e_flags bits is of course vastly simpler to deal with than looking for an ELF note, so if the bit is potentially forward looking I'm happy with having a flag in e_flags.

Reply:
It is difficult to make predictions at this stage. We can either reuse or rename the flag for a future architecture.
However we don't have any strong opinions on the name so will go with the more generic `EF_AARCH64_CHERI_PURECAP`.

Reply:
@jrtc27, @bsdjhb
Are you happy with these changes?